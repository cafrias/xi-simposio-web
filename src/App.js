// @flow

import React, {Component} from 'react'
import './App.sass'

// COMPONENTS __________________________________________________________________
import Navbar from './components/Navbar'
import ToTop from './components/ToTop'

// VIEWS _______________________________________________________________________
import Home from './views/Home'
import Speakers from './views/Speakers'
import About from './views/About'
import Schedule from './views/Schedule'
import Registration from './views/Registration'
import Location from './views/Location'

class App extends Component {
  state: {
    menuOpen: boolean,
    navigateTo: string
  }

  constructor(props: any) {
    super(props)

    this.state = {
      menuOpen: false,
      navigateTo: ''
    }

    ;(this:any).toggleMenu = this.toggleMenu.bind(this)
    ;(this:any).closeAndNavigate = this.closeAndNavigate.bind(this)
  }

  toggleMenu(): void {
    this.setState({
      menuOpen: !this.state.menuOpen,
    })
  }

  closeAndNavigate(hashName: string): void {
    this.setState({
      menuOpen: false,
      navigateTo: hashName
    })
  }

  componentDidUpdate() {
    const {
      navigateTo
    } = this.state

    if(navigateTo) {
      location.hash = navigateTo
      this.setState({
        navigateTo: ''
      })
    }
  }

  render() {
    const {
      menuOpen
    } = this.state

    return (
      <div className="app">
        <Navbar menuOpen={menuOpen} closeAndNavigate={this.closeAndNavigate} toggleMenu={this.toggleMenu} />
        <main className={`content-container ${menuOpen ? 'no-scroll' : ''}`}>
          <Home/>
          <About/>
          <Speakers/>
          <Schedule/>
          <Registration/>
          <Location/>
        </main>
        <ToTop/>
        <footer className="main-footer">
          <small>Design and development – Frias Design®</small>
        </footer>
      </div>
    )
  }
}

export default App
