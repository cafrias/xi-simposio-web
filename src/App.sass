@import './_sass/_breakpoints.sass'

.content-container
  margin-top: 5.295rem
  max-width: 100vw

.no-scroll
  height: 10px
  overflow: hidden

+for-desktop-up
  .content-container
    margin-top: 6.854rem
