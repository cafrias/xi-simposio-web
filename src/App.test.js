import React from 'react'
import {shallow} from 'enzyme'

import App from './App'

describe('App view', () => {
  var wrapper

  beforeEach(() => wrapper = shallow(<App />))

  it('should render without errors', () => {
    expect(wrapper).toHaveLength(1)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
