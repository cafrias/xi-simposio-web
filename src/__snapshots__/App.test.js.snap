exports[`App view should match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="app">
    <Navbar
        closeAndNavigate={[Function]}
        menuOpen={false}
        toggleMenu={[Function]} />
    <main
        className="content-container ">
        <Home />
        <About />
        <Speakers />
        <Schedule />
        <Registration />
        <Location />
    </main>
    <ToTop />
    <footer
        className="main-footer">
        <small>
            Design and development – Frias Design®
        </small>
    </footer>
</div>,
  "nodes": Array [
    <div
      className="app">
      <Navbar
            closeAndNavigate={[Function]}
            menuOpen={false}
            toggleMenu={[Function]} />
      <main
            className="content-container ">
            <Home />
            <About />
            <Speakers />
            <Schedule />
            <Registration />
            <Location />
      </main>
      <ToTop />
      <footer
            className="main-footer">
            <small>
                  Design and development – Frias Design®
            </small>
      </footer>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "closeAndNavigate": [Function],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "menuOpen": false,
          "navigateTo": "",
        },
        "toggleMenu": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="app">
          <Navbar
                    closeAndNavigate={[Function]}
                    menuOpen={false}
                    toggleMenu={[Function]} />
          <main
                    className="content-container ">
                    <Home />
                    <About />
                    <Speakers />
                    <Schedule />
                    <Registration />
                    <Location />
          </main>
          <ToTop />
          <footer
                    className="main-footer">
                    <small>
                              Design and development – Frias Design®
                    </small>
          </footer>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="app">
          <Navbar
                    closeAndNavigate={[Function]}
                    menuOpen={false}
                    toggleMenu={[Function]} />
          <main
                    className="content-container ">
                    <Home />
                    <About />
                    <Speakers />
                    <Schedule />
                    <Registration />
                    <Location />
          </main>
          <ToTop />
          <footer
                    className="main-footer">
                    <small>
                              Design and development – Frias Design®
                    </small>
          </footer>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
